Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -81,7 +82,9 @@ public SecurityChanges  ApplyUniverseSelection(Universe universe, DateTime dateT
                     var fineCollection = new BaseDataCollection();
                     var dataProvider = new DefaultDataProvider();
 
-                    foreach (var symbol in selectSymbolsResult)
+                    // use all available threads, the entire system is waiting for this to complete
+                    var options = new ParallelOptions{MaxDegreeOfParallelism = Environment.ProcessorCount};
+                    Parallel.ForEach(selectSymbolsResult, options, symbol =>
```

Comment:
😎

Reply:
Beautiful! just out of curiosity, did you test using `> Environment.ProcessorCount`? Like `Environment.ProcessorCount * 1.5`, Maybe there is room for even more juice

Reply:
I did not try but I can. I wouldn't expect that to help since all of these reads are synchronous -- if we were doing async reads I would think we could squeeze a little more out of it

Reply:
@Martin-Molinero -- tried 1.5x and didn't notice and improvements (statistically insignificant delay was actually observed) -- hunch is that it's just extra overhead and the threads are simply competing each other